src/main/java/com/codejoust/main/model/report/GameReport.java

```diff
@@ -0,0 +1,55 @@
+package com.codejoust.main.model.report;
+
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.persistence.Entity;
+import javax.persistence.FetchType;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.OneToMany;
+
+import com.codejoust.main.model.User;
+import com.codejoust.main.model.problem.Problem;
+
+import org.hibernate.annotations.Fetch;
+import org.hibernate.annotations.FetchMode;
+
+import lombok.AccessLevel;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+
+@EqualsAndHashCode(onlyExplicitlyIncluded = true)
+@Entity
+@Getter
+@Setter
+public class GameReport {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.AUTO)
+    private Integer id;
+
+    private String gameReportId;
+
+    @OneToMany(mappedBy = "gameReport", fetch = FetchType.EAGER)
+    @Setter(AccessLevel.PRIVATE)
+    @Fetch(value = FetchMode.SUBSELECT)
+    private List<Problem> problems = new ArrayList<>();
+
+    @OneToMany(mappedBy = "gameReport", fetch = FetchType.EAGER)
+    @Setter(AccessLevel.PRIVATE)
```

Comment:
Same question repeated here - should these users and problems fields be ManyToMany?

Reply:
You're right, they absolutely should. Once I update the `problems` to the `ProblemContainer` type though, that should have a `OneToMany` mapping because there is a new `ProblemContainer` for each new game.